src/modules/site-settings/edit-site-details.tsx

```diff
@@ -52,8 +53,25 @@ export default function EditSiteDetails( { currentWpVersion, onSave }: EditSiteD
 		label: version.label,
 		value: version.value,
 	} ) );
+
 	if ( ! wordpressVersionOptions.some( ( version ) => version.value === currentWpVersion ) ) {
-		wordpressVersionOptions.push( { label: currentWpVersion, value: currentWpVersion } );
+		const customVersion = { label: currentWpVersion, value: currentWpVersion };
+		// find the index of the version that is less than the current version
+		const insertIndex = wordpressVersionOptions.findIndex( ( compareVersion ) => {
```

Comment:
What about this, to avoid comment and make variable more obvious to "what it is", i stead of "what it should do":
```suggestion
		const indexOfPrevVersion = wordpressVersionOptions.findIndex( ( compareVersion ) => {
```

Reply:
Also, let's extract this insertion to a separate function. Ar a result the component will be smaller and easy to read.

```
if ( ! wordpressVersionOptions.some( ( version ) => version.value === currentWpVersion ) ) {
	addWpVersionToList( currentWpVersion, wordpressVersionOptions );
}
```